src/helpers/references.ts

```diff
@@ -58,8 +58,16 @@ export function isSingleCellReference(xc: string): boolean {
 }
 
 export function splitReference(ref: string): { sheetName?: string; xc: string } {
+  if (!ref.includes("!")) {
+    return { xc: ref };
+  }
   const parts = ref.split("!");
   const xc = parts.pop()!;
   const sheetName = getUnquotedSheetName(parts.join("!")) || undefined;
   return { sheetName, xc };
 }
+
+/** Return a reference SheetName!xc from the given arguments */
+export function getReference(xc: string, sheetName?: string): string {
+  return sheetName ? `${sheetName}!${xc}` : xc;
```

Comment:
I think it would be safer to use `getCanonicalSheetName`. I don't think we have any guarantees  it's already canonical.
And some uses of `getCanonicalSheetName` can probably be changed to use this function